openretina/insilico/tuning_analyses/response_gradient.py

```diff
@@ -0,0 +1,110 @@
+
+    Args:
+        stim (torchTensor, 1 x n_channels x time x height x width): The MEI stimulus.
+        model_neuron (SingleNeuronObjective): The model neuron objective.
+        n_channels (int, optional): The number of channels. Defaults to 2.
+        start (float, optional): The starting value for the contrast range. Defaults to -1.
+        stop (float, optional): The ending value for the contrast range. Defaults to 1.
+        step_size (float, optional): The step size for the contrast range. Defaults to 0.1.
+
+    Returns:
+        tuple: A tuple containing the following elements:
+            - grid (ndarray): A grid of gradient values with shape (n_channels, len(green_contrast_values), len(uv_contrast_values)).
+            - resp_grid (ndarray): A grid of loss values with shape (len(green_contrast_values), len(uv_contrast_values)).
+            - norm_grid (ndarray): A grid of norm values with shape (len(green_contrast_values), len(uv_contrast_values)).
+            - green_contrast_values (ndarray): An array of green contrast values.
+            - uv_contrast_values (ndarray): An array of UV contrast values.
+    """
+    green_contrast_values = np.arange(start, stop+step_size, step_size)
+    uv_contrast_values = np.arange(start, stop+step_size, step_size)
+    grid = np.zeros((n_channels, len(green_contrast_values), len(uv_contrast_values)))
+    resp_grid = np.zeros((len(green_contrast_values), len(uv_contrast_values)))
+    norm_grid = np.zeros((len(green_contrast_values), len(uv_contrast_values)))
+
+    for i, contrast_green in enumerate(np.arange(-1, 1+step_size, step_size)):
+        for j, contrast_uv in enumerate(np.arange(-1, 1+step_size, step_size)):
+            mei_contrast_gen = MeiAcrossContrasts(torch.Tensor([contrast_green, contrast_uv]), stim)
+            response_gradient, response = trainer_fn(mei_contrast_gen, model_neuron, lr=.1)
+            grid[0, i, j] = response_gradient[0]
+            grid[1, i, j] = response_gradient[1]
+            resp_grid[i, j] = response
+            norm_grid[i, j] = np.linalg.norm(grid[:, i, j])
+    return grid, resp_grid, norm_grid, green_contrast_values, uv_contrast_values
+
+
+def equalize_channels(mei: torch.Tensor,
+                      flip_green: bool = False) -> torch.Tensor:
+    '''
+    Scales the (green and UV) channels of an MEI such that they have equal norm,
+    and the scaled MEI has the same norm as the original MEI. Optionally flips sign of green channel
+    :param mei: torch.Tensor, shape (1, 2, 50, 18, 16)
+    :param flip_green: bool
+    :return: equalized_mei: torch.Tensor, shape (1, 2, 50, 18, 16)
+    '''
+    green_chan = mei[0, 0]
+    uv_chan = mei[0, 1]
+    green_norm = torch.norm(green_chan)
+    uv_norm = torch.norm(uv_chan)
+    total_norm = torch.norm(mei)
+    green_factor = (total_norm/2)/green_norm
+    uv_factor = (total_norm/2)/uv_norm
+    equalized_mei = torch.zeros_like(mei).cuda()
```

Comment:
```suggestion
    equalized_mei = torch.zeros_like(mei, device=mei.device)
```
I think if you change the code like this it will work on both cpu and gpu (I think it's best if we strive to have no explicit .cuda() calls in our code except there is a good reason to have them :) )